hs_core/models.py

```diff
@@ -1127,11 +1128,17 @@ def delete(self, using=None):
         self.metadata.delete_all_elements()
         self.metadata.delete()
 
+        # delete related access controls records
         access_resource = self.raccess
         UserResourcePrivilege.objects.filter(resource=access_resource).delete()
         GroupResourcePrivilege.objects.filter(resource=access_resource).delete()
         access_resource.delete()
 
+        # delete related resource labelling records
+        res_labels = self.rlabels
+        UserResourceLabels.objects.filter(rlabels=res_labels).delete()
+        res_labels.delete()
```

Comment:
This should not be needed. If the labels have a reference to the resource, they'll be cascade deleted automatically.